Sage/carbon
# chore(heading): remove classic theme

removal of the classic theme styling from the heading component. Old documentation has been replaced
with a CSF stories file and an MDX example of the component.

BREAKING CHANGE: removal of the classic styling from the heading component.

fixes FE-3109

### Proposed behaviour
- Removes `classic` theme from the Heading component.
- Update documentation to be inline with our current standards.
- Update stories file to be in CSF.

### Current behaviour
- Heading component currently supports `classic` theme.
- Documentation adheres to our previous standards.
- stories file is in the depreciated `storiesOf` format.

### Checklist



### Additional context
N/A

### Testing instructions
- `git checkout FE-3109-remove_classic_theme_heading`
- Start Storybook using `npm start`
- Navigate to the Heading component
- Heading component should not have any `classic` styling
- Heading documentation should be in an MDX example
